src/Aspire.Dashboard/Components/Pages/Resources.razor.cs

```diff
@@ -164,6 +164,16 @@ private async Task OnResourceListChanged(ObjectChangeType objectChangeType, Reso
         await InvokeAsync(StateHasChanged);
     }
 
+    private string GetResourceName(ResourceViewModel resource)
+    {
+        if (_resourcesMap.Count(kvp => kvp.Value.DisplayName == resource.DisplayName) >= 2)
```

Comment:
`Enumerable.Count` will enumerate more than needed in some cases. As soon as the second match is found, it could return. Can this be made similar to the method you added in `OtlpApplication.GetResourceName`? Or use `Skip(1).Any()`.

Reply:
`Skip(1).Any()`: cute

I'll repeat `OtlpApplication.GetResourceName` but for the resource view model. It avoids over enumerating and allocating an enumerator.